lib/FileSystemInfo.js

```diff
@@ -1685,7 +1685,7 @@ class FileSystemInfo {
 													type: RBDT_RESOLVE_ESM_FILE,
 													context,
 													path: dependency,
-													expected: undefined,
+													expected: false,
```

Comment:
Only set this to `false` for the `import()` case, which is the `if (imp.d > -1)` case a few lines above.